Zend/zend_enum.c

```diff
@@ -180,14 +180,16 @@ void zend_enum_add_interfaces(zend_class_entry *ce)
 
 	ZEND_ASSERT(!(ce->ce_flags & ZEND_ACC_RESOLVED_INTERFACES));
 
-	ce->interface_names = erealloc(ce->interface_names, sizeof(zend_class_name) * ce->num_interfaces);
+	ce->interface_names = erealloc(ce->interface_names, sizeof(zend_interface_name) * ce->num_interfaces);
```

Comment:
This is a change that should definitely be in UPGRADING.INTERNALS — I know of several extensions/prototypes that I have written where this would start silently causing issues.

Reply:
It definitely should be mentioned in the UPGRADING.INTERNALS, but the correct solution would nevertheless be the following:

```suggestion
	ce->interface_names = safe_erealloc(ce->interface_names, ce->num_interfaces, sizeof(*ce->interface_names), 0);
```

Thus not hardcoding the struct name in the first place.